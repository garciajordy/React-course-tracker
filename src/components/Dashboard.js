import React, { useEffect, useState } from 'react';
import axios from "axios";
import store from "../store/Store";
import { useHistory } from 'react-router-dom';
import logIn from "../actions/LoginAction";
import Course from "./Course";
import AddCourseAction from "../actions/AddCourseAction";

export const COURSES = ["RubyOnRails", "Ruby", "JavaScript", "React"]
const Dashboard = () => {
    let history = useHistory()
    const [courses, setCourses] = useState([])
    useEffect(() => {
        checkLoginStatus()
        axios.get("http://localhost:3001/courses").then(response => {
                setCourses(response.data.map(data => data))
            })
    },[])
    function checkLoginStatus() {
        axios.get("http://localhost:3001/logged_in", { withCredentials: true }).then(response => {
            console.log(response)
            if (response.data.logged_in) {
                store.dispatch(logIn(response.data.user))
            } else {
                history.push("/")
                console.log("heeeeeeeee",response)
            }
        }).catch(error => {
            history.push("/")
          console.log(error)
        })
      }
    const clickHandler = (course) => {
        store.dispatch(AddCourseAction(course))
        console.log(course)
      }
    return (
        <div>
            <h1 className="dashboard">Dashboard</h1>
            <div className="course-container">

            {courses.map(course => (
                <Course key={course.id} handleClick={clickHandler} course={course} />
            ))}
            </div>
        </div>
    )
}

export default Dashboard
