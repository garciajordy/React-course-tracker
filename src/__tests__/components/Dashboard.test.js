import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Dashboard from '../../components/Dashboard';
import store from '../../store/Store';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><Dashboard /></Provider>);
  expect(asFragment(<Provider store={store}><Dashboard /></Provider>)).toMatchSnapshot();
});

it('displays heading', () => {
  render(<Provider store={store}><Dashboard /></Provider>);
  expect(screen.getByText('Dashboard')).toBeTruthy();
});
