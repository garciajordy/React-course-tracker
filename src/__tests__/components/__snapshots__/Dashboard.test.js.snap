// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="header"
    >
      All courses.it
    </div>
    <h1
      class="dashboard"
    >
      Dashboard
    </h1>
    <div
      class="course-container"
    />
  </div>
</DocumentFragment>
`;
